test/Microsoft.AspNetCore.Mvc.Core.Test/Controllers/DefaultControllerTypeProviderTest.cs

```diff
@@ -230,18 +230,31 @@ public void IsController_NoControllerSuffix()
         [Theory]
         [InlineData(typeof(DescendantLevel1))]
         [InlineData(typeof(DescendantLevel2))]
-        public void IsController_ReturnsTrue_IfAncestorTypeNameHasControllerSuffix(Type type)
+        public void IsController_ReturnsTrue_IfAncestorTypeHasControllerAttribute(Type type)
         {
             // Arrange
             var provider = GetControllerTypeProvider();
 
             // Act
-            var isController = provider.IsController(type.GetTypeInfo(), CandidateAssemblies);
+            var isController = provider.IsController(type.GetTypeInfo());
 
             // Assert
             Assert.True(isController);
         }
 
+        [Fact]
+        public void IsController_ReturnsFalse_IfAncestorTypeDoesNotHaveControllerAttribute()
+        {
+            // Arrange
+            var provider = GetControllerTypeProvider();
+
+            // Act
+            var isController = provider.IsController(typeof(NoSuffixNoControllerAttribute).GetTypeInfo());
+
+            // Assert
+            Assert.False(isController);
+        }
+
         [Theory]
         [InlineData(typeof(BaseNonControllerController))]
         [InlineData(typeof(BaseNonControllerControllerChild))]
```

Comment:
Need a few more tests involving `[NonController]` e.g. a class w/ `[Controller]` and suffix "Controller" that inherits from something w/ `[NonController]`.